extensions/amp-viz-vega/0.1/amp-viz-vega.js

```diff
@@ -0,0 +1,163 @@
+  }
+
+  /** @override */
+  buildCallback() {
+    user.assert(isExperimentOn(this.win, EXPERIMENT),
+      `Experiment ${EXPERIMENT} disabled`);
+
+    /** @private {?JSONType} */
+    this.data_ = null;
+
+    /** @private {?string} */
+    this.inlineData_ = this.getInlineData_();
+
+    /** @private {?string} */
+    this.dataUrl_ = this.element.getAttribute('data-url');
+
+    user.assert(this.inlineData_ || this.dataUrl_,
+      '%s: neither `data-url` attribute nor a ' +
+      'valid <script type="application/json"> child was found for Vega data.',
+      this.getName_());
+
+    user.assert(!(this.inlineData_ && this.dataUrl_),
+      '%s: both `data-url` attribute and a valid ' +
+      '<script type="application/json"> child were found for Vega data. ' +
+      'Only one way of specifying the data is allowed.',
+      this.getName_());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.initialize_();
+    return this.loadData_().then(() => this.renderGraph_());
+  }
+
+  /**
+   * Create the vega container.
+   * Called lazily in the first `#layoutCallback`.
+   */
+  initialize_() {
+    if (this.container_) {
+      return;
+    }
+
+    /** @private {?Element} */
+    this.container_ = this.element.ownerDocument.createElement('div');
+
+    this.applyFillContent(this.container_, true);
+    this.element.appendChild(this.container_);
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  loadData_() {
+    if (this.data_) {
+      return Promise.resolve();
+    }
+
+    // Validation in buildCallback should ensure one and only one of
+    // dataUrl_/inlineData_ is ever set.
+    dev.assert(!this.dataUrl_ != !this.inlineData_);
+
+    if (this.inlineData_) {
+      this.data_ = tryParseJson(this.inlineData_, err => {
+        user.assert(!err, 'data could not be ' +
+        'parsed. Is it in a valid JSON format?: %s', err);
+      });
+    } else {
+      // TODO(aghassemi): Fetch and validate the data file.
+      this.data_ = {
+        'url': this.dataUrl_,
+      };
+    }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * @return {?string|undefined}
+   * @private
+   */
+  getInlineData_() {
+    let inlineConfig;
+
+    const scripts = childElementsByTag(this.element, 'SCRIPT');
+    if (scripts.length == 1) {
+      const child = scripts[0];
+      if (isJsonScriptTag(child)) {
+        inlineConfig = child.textContent;
+      } else {
+        user.error(this.getName_(), 'data should ' +
+          'be put in a <script type="application/json"> tag.');
+      }
+    } else if (scripts.length > 1) {
+      user.error(this.getName_(), 'more than one' +
+        '<script type="application/json"> tags found. Only one allowed.');
+    }
+    return inlineConfig;
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  renderGraph_() {
+    // TODO(aghassemi): Replace with actual rendering implementation.
+    return new Promise((resolve, unused) => {
+      setTimeout(() => {
```

Comment:
The render code is temp of course, had the timeout to delay the rendering mostly to make sure placeholder shows up fine. Removed it for now. Regardless, render code will change when Vega rendering is integrated.